cdap-common/src/main/java/co/cask/cdap/common/namespace/DefaultNamespacedLocationFactory.java

```diff
@@ -49,57 +48,43 @@ public DefaultNamespacedLocationFactory(CConfiguration cConf,
-    return get(namespaceId, null);
+  public Location get(NamespaceId namespaceId) throws IOException {
+    if (NamespaceId.DEFAULT.equals(namespaceId)
+      || NamespaceId.SYSTEM.equals(namespaceId)
+      || NamespaceId.CDAP.equals(namespaceId)) {
+
+      // since these are cdap reserved namespace we know there cannot be a custom mapping for this.
+      // for optimization don't query for namespace meta
+      return getNonCustomMappedLocation(namespaceId);
+    }
+    // since this is not a cdap reserved namespace we look up meta if there is a custom mapping
+    try {
+      return get(namespaceQueryAdmin.get(namespaceId));
+    } catch (IOException e) {
+      throw e;
+    } catch (Exception e) {
+      throw new IOException(String.format("Failed to get namespace meta for namespace %s", namespaceId), e);
+    }
   }
 
   @Override
   public Location get(NamespaceMeta namespaceMeta) throws IOException {
     String rootDirectory = namespaceMeta.getConfig().getRootDirectory();
     if (Strings.isNullOrEmpty(rootDirectory)) {
       // if no custom mapping was specified the use the default namespaces location
-      return getNonCustomMappedLocation(namespaceMeta.getNamespaceId().toId());
+      return getNonCustomMappedLocation(namespaceMeta.getNamespaceId());
     }
     return Locations.getLocationFromAbsolutePath(locationFactory, rootDirectory);
   }
 
-  @Override
-  public Location get(Id.Namespace namespaceId, @Nullable String subPath) throws IOException {
-    Location namespaceLocation;
-    if (Id.Namespace.DEFAULT.equals(namespaceId) || Id.Namespace.SYSTEM.equals(namespaceId) ||
-      Id.Namespace.CDAP.equals(namespaceId)) {
-      // since these are cdap reserved namespace we know there cannot be a custom mapping for this.
-      // for optimization don't query for namespace meta
-      namespaceLocation = getNonCustomMappedLocation(namespaceId);
-    } else {
-      // since this is not a cdap reserved namespace we look up meta if there is a custom mapping
-      NamespaceMeta namespaceMeta;
-      try {
-        namespaceMeta = namespaceQueryAdmin.get(namespaceId.toEntityId());
-      } catch (Exception e) {
-        throw new IOException(String.format("Failed to get namespace meta for namespace %s", namespaceId), e);
-      }
-      namespaceLocation = get(namespaceMeta);
-    }
-
-    if (subPath != null) {
-      namespaceLocation = namespaceLocation.append(subPath);
-    }
-    return namespaceLocation;
-  }
-
   /**
    * Gives a namespaced location for a namespace which does not have a custom mapping
    *
    * @param namespaceId the simple cdap namespace
    * @return location of the namespace
    */
-  private Location getNonCustomMappedLocation(Id.Namespace namespaceId) throws IOException {
-    return locationFactory.create(namespaceDir).append(namespaceId.getId());
+  private Location getNonCustomMappedLocation(NamespaceId namespaceId) throws IOException {
```

Comment:
the `locationFactory.create` is not creating anything physical. A factory.create method is just to create an object instance.